tensorflow/compiler/mlir/hlo/runtime/ral/ral_logging.h

```diff
@@ -0,0 +1,133 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_MLIR_HLO_RUNTIME_RAL_RAL_LOGGING_H_
+#define TENSORFLOW_COMPILER_MLIR_HLO_RUNTIME_RAL_RAL_LOGGING_H_
+
+#include <atomic>
+#include <limits>
+#include <memory>
+#include <sstream>
+
+namespace mlir {
+namespace disc_ral {
+
+// -------------------------------------------------------------------
+
+// This file contains a light version of Logger implemented in
+// tensorflow/core/platform/default/logging.h
+//
+// We re-implement it here because we do not want to rely
+// on TensorFlow data structures, and hence we can move
+// corresponding implementations to llvm repo. easily in case necessary.
```

Comment:
Is the runtime intended to have zero dependency or do you plan to depend on LLVM libSupport at some point?

Asking because you're not using StringRef and many other ADT right now.

Reply:
Yes. We suppose this part of code will be integrated to the AI framework (e.g. TF, PyTorch) plugin. Thus, it'd be better to not introduce other dependency at the moment.

Reply:
That make sense, but can you then not hint at using LLVM implementation in the comment here and instead mention that we'd like to have a runtime with no extra dependencies.

It may be worth also adding this into a README.md in the runtime directory.